presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -168,7 +168,8 @@ public ConnectorPageSource createPageSource(
                 split.getLength(),
                 split.getFileFormat(),
                 regularColumns,
-                table.getPredicate());
+                table.getPredicate(),
+                dynamicFilter.transform(IcebergColumnHandle.class::cast));
```

Comment:
Please change the argument of `ConnectorPageSource#createPageSource` from `TupleDomain<ColumnHandle> dynamicFilter` to `Dynamic dynamicFilter`.
Here we can do 
```
table.getPredicate().intersect(dynamicFilter.getCurrentPredicate().transform(IcebergColumnHandle.class::cast).simplify(100))
```
That should remove the need for the rest of the changes in this file.